### PR TITLE
New module: Add monitoring/zabbix_template module.

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY
Implements zabbix_template module to manipulate zabbix template objects on zabbix server via API.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
modules/monitoring/zabbix_template

##### ANSIBLE VERSION
```
ansible 2.4.0
  config file =
  configured module search path = [u'/Users/sookido/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible-2.4.0-py2.7.egg/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Apr  4 2017, 08:47:57) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]

```

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: lib/ansible/modules/monitoring/zabbix_template.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+    template = Template(module, zbx)
+    template_ids = template.get_template_ids([template_name])
+
+    # delete template
+    if state == "absent":
+        # if template not found. no change, no fail
+        if not template_ids:
+            module.exit_json(changed=False,
+                             result="Template not found. " +
```
Comment (resmo) [inline lib/ansible/modules/monitoring/zabbix_template.py]: would you mind replacing the keys `result` by `msg` by convention?

File: lib/ansible/modules/monitoring/zabbix_template.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    def get_template_ids(self, template_list):
+        template_ids = []
+        if template_list is None or len(template_list) == 0:
+            return template_ids
+        for template in template_list:
+            template_list = self._zapi.template.get(
+                {'output': 'extend',
+                 'filter': {'host': template}})
+            if len(template_list) < 1:
+                return None
```
Comment (resmo) [inline lib/ansible/modules/monitoring/zabbix_template.py]: Do you really want to return here or should this not just `continue` with the next iteration?

**Review by resmo — COMMENTED** [review]
good job. I have a small change request and a question

**Comment by ghost** [comment]
@resmo 
Thank you for your prompt reply

fix some bugs
- exit_json argument
- loop bug

**Comment by resmo** [comment]
shipit

**Comment by ghost** [comment]
please shipit!

**Comment by bhcopeland** [comment]
shipit

**Comment by eikef** [comment]
This works fine in testing, but is not idempotent -- it will always have changed: True unless you are trying to delete a template that is not there.
This can be added in later though if somebody wants to implement it.

The TLS-changes in the other modules also need to be ported here, but that can also happen after it is included.

**Comment by eikef** [comment]
shipit

**Comment by eikef** [comment]
needs_revision

File: lib/ansible/modules/monitoring/zabbix_template.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
```
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_template.py]: Please change this to 1.1

File: lib/ansible/modules/monitoring/zabbix_template.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: zabbix_template
+short_description: create/delete zabbix template
+description:
+    - create/delete zabbix template
+version_added: "2.4"
```
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_template.py]: Please change this to 2.5

File: lib/ansible/modules/monitoring/zabbix_template.py [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_template.py]: please add 
```
from __future__ import absolute_import, division, print_function
__metaclass__ = type

```

**Review by eikef — CHANGES_REQUESTED** [review]
With these three changes, this should be committable :-)

**Comment by ghost** [comment]
@eikef
Thank you for your reply
update ansible 2.5

**Review by eikef — APPROVED** [review]

**Comment by eikef** [comment]
shipit

**Comment by eikef** [comment]
!needs_revision

**Comment by eikef** [comment]
bot_status

**Comment by resmo** [comment]
shipit

**Comment by resmo** [comment]
thanks!

**Pull request merged**